tests/tests_integration/test_api/test_workflows.py

```diff
@@ -0,0 +1,348 @@
+                        is_async_complete=True,
+                    ),
+                    timeout=120,
+                    retries=2,
+                ),
+            ],
+        ),
+    )
+    retrieved = cognite_client.workflows.versions.retrieve(workflow_id, version.version)
+    if retrieved is not None:
+        return retrieved
+    return cognite_client.workflows.versions.upsert(version)
+
+
+@pytest.fixture
+def workflow_execution_list(
+    cognite_client: CogniteClient, add_multiply_workflow: WorkflowVersion
+) -> WorkflowExecutionList:
+    executions = cognite_client.workflows.executions.list(workflow_ids=add_multiply_workflow.as_id(), limit=5)
+    if executions:
+        return executions
+    # Creating at least one execution
+    result = cognite_client.workflows.executions.trigger(
+        add_multiply_workflow.workflow_external_id, add_multiply_workflow.version, {"a": 5, "b": 6}
+    )
+    t0 = time.time()
+    while result.status != "completed":
+        result = cognite_client.workflows.executions.retrieve_detailed(result.id)
+        cognite_client.workflows.tasks.update(result.executed_tasks[1].id, "completed")
+        time.sleep(5)
+        if time.time() - t0 > 60:
+            raise TimeoutError("Workflow execution did not complete in time")
+    return cognite_client.workflows.executions.list(workflow_ids=add_multiply_workflow.as_id(), limit=5)
+
+
+class TestWorkflows:
+    def test_upsert_delete(self, cognite_client: CogniteClient) -> None:
+        workflow = WorkflowCreate(
+            external_id="integration_test-test_create_delete",
+            description="This is ephemeral workflow for testing purposes",
+        )
+        cognite_client.workflows.delete(workflow.external_id, ignore_unknown_ids=True)
+
+        created_workflow: Workflow | None = None
+        try:
+            created_workflow = cognite_client.workflows.upsert(workflow)
+
+            assert created_workflow.external_id == workflow.external_id
+            assert created_workflow.description == workflow.description
+            assert created_workflow.created_time is not None
+        finally:
+            if created_workflow is not None:
+                cognite_client.workflows.delete(created_workflow.external_id)
+
+    def test_delete_non_existing_raise(self, cognite_client: CogniteClient) -> None:
+        with pytest.raises(CogniteAPIError) as e:
+            cognite_client.workflows.delete("integration_test-non_existing_workflow", ignore_unknown_ids=False)
+
+        assert "workflows were not found" in str(e.value)
+
+    def test_delete_non_existing(self, cognite_client: CogniteClient) -> None:
+        cognite_client.workflows.delete("integration_test-non_existing_workflow", ignore_unknown_ids=True)
+
+    def test_list_workflows(self, cognite_client: CogniteClient, workflow_list: WorkflowList) -> None:
+        listed = cognite_client.workflows.list()
+
+        assert len(listed) >= len(workflow_list)
```

Comment:
I vote to skip checks like this. When we run in CI, the windows runner and linux runner will make this flaky.